File: src/main/java/net/alephdev/calendar/models/Sprint.java
package net.alephdev.calendar.models;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "Sprint")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Sprint {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false)
    private String majorVersion;

    @Column(nullable = false)
    private LocalDate startDate;

    @Column(nullable = false)
    private LocalDate endDate;

    @Column(nullable = false)
    private LocalDate regressionStart;

    @Column(nullable = false)
    private LocalDate regressionEnd;

    @ManyToOne
    @JoinColumn(name = "team_id", foreignKey = @ForeignKey(name = "fk_sprint_team_id"), nullable = false)
    private Team team;


}

File: angular/src/app/models/sprint.ts
import {Team} from "./team";

export interface Sprint {
  id: number;
  majorVersion: string;
  startDate: string; // Date in ISO format
  endDate: string; // Date in ISO format
  regressionStart: string; // Date in ISO format
  regressionEnd: string; // Date in ISO format
  team: Team;
}
